spec/controllers/manage_evaluators_controller_spec.rb

```diff
@@ -84,27 +84,91 @@
+        expect {
+          post challenge_manage_evaluators_path(challenge), params: {
+            evaluator_invitation: {
+              email: existing_user.email,
+              phase_id: phase.id
+            }
           }
-        }
+        }.not_to change(ChallengePhasesEvaluator, :count)
+
+        expect(ChallengePhasesEvaluator.where(challenge: challenge, phase: phase, user: existing_user).count).to eq(1)
+        expect(EvaluatorInvitation.find_by(id: invitation.id)).to be_present
+        expect(response).to redirect_to(challenge_manage_evaluators_path(challenge, phase_id: phase.id))
+        expect(flash[:notice]).to include("has already been added as an evaluator for this phase")
       end
+    end
 
-      it 'creates a new evaluator invitation' do
+    context 'when adding an existing user with a non-evaluator role' do
+      let(:existing_user) { create(:user, role: 'solver', status: 'pending') }
+
+      it 'adds the user as an evaluator without changing their role' do
         expect {
-          post challenge_manage_evaluators_path(challenge), params: new_user_params
-        }.to change(EvaluatorInvitation, :count).by(1)
+          post challenge_manage_evaluators_path(challenge), params: {
+            evaluator_invitation: {
+              email: existing_user.email,
+              phase_id: phase.id
+            }
+          }
+        }.to change(ChallengePhasesEvaluator, :count).by(1)
+
+        existing_user.reload
+        expect(existing_user.role).to eq('solver')
+        expect(response).to redirect_to(challenge_manage_evaluators_path(challenge, phase_id: phase.id))
+        expect(flash[:notice]).to include("has been added as an evaluator for this phase")
+      end
+    end
+
+    context 'when adding a new user with default evaluator role' do
+      let(:existing_evaluator) { create(:user, role: 'evaluator', status: 'pending') }
+
+      it 'adds the user without changing their role' do
+        expect {
+          post challenge_manage_evaluators_path(challenge), params: {
+            evaluator_invitation: {
+              email: existing_evaluator.email,
+              phase_id: phase.id
+            }
+          }
+        }.to change(ChallengePhasesEvaluator, :count).by(1)
 
-        expect(ChallengePhasesEvaluator.count).to eq(0)
+        existing_evaluator.reload
+        expect(existing_evaluator.role).to eq('evaluator')
         expect(response).to redirect_to(challenge_manage_evaluators_path(challenge, phase_id: phase.id))
-        expect(flash[:notice]).to include("Invitation sent to")
+        expect(flash[:notice]).to include("has been added as an evaluator for this phase")
+      end
+    end
+
+    context 'when adding an existing user with an invalid role' do
+      let(:existing_user) { create(:user, role: 'evaluator', status: 'pending') }
+
+      before do
+        existing_user.update_column(:role, 'admin') # invalid evaluator role
+      end
```

Comment:
can't you create the user as an admin instead of updating them after create?
```suggestion
      let(:existing_user) { create(:user, role: 'admin', status: 'active') }
```